pkg/core/validation/format.go

```diff
@@ -50,37 +51,41 @@ type URL struct {
 	URLVariant URLVariant
 }
 
-// nolint: gocyclo
 func (f URL) IsFormat(input interface{}) bool {
+	return f.ValidateFormat(input) == nil
+}
+
+// nolint: gocyclo
+func (f URL) ValidateFormat(input interface{}) error {
 	str, ok := input.(string)
 	if !ok {
-		return false
+		return errors.New("input must be a string")
 	}
 	if str == "" {
-		return false
+		return errors.New("input must not be empty")
 	}
 
 	u, err := url.Parse(str)
 	if err != nil {
-		return false
+		return err
 	}
 	if u.RawQuery != "" || u.Fragment != "" {
-		return false
+		return errors.New("input URL must not have query/fragment")
 	}
 
 	p := ""
 	switch f.URLVariant {
 	case URLVariantFullOnly:
 		if u.Scheme == "" || u.Host == "" {
-			return false
+			return errors.New("input URL must be complete")
```

Comment:
Maybe `must be absolute` is a more accurate term?

Reply:
Sure. I'll then use `input URL must be normalized` for the `filepath.Clean` check below.

Reply:
I mean in [RFC3986](https://tools.ietf.org/html/rfc3986#section-4.3) absolute URI is defined and its meaning matches the case here. The error message should be `input URL must be absolute`